src/main/java/org/jenkinsci/plugins/ghprb/GhprbTrigger.java

```diff
@@ -372,11 +370,10 @@ public String getGitHubAuthId() {
     
     public GhprbGitHubAuth getGitHubApiAuth() {
         if (gitHubAuthId == null) {
-            for (GhprbGitHubAuth auth: getDescriptor().getGithubAuth()){
-                gitHubAuthId = auth.getId();
-                getDescriptor().save();
-                return auth;
-            }
+            GhprbGitHubAuth auth = getDescriptor().getGithubAuth().get(0);
```

Comment:
This assumes that there will always be at least one entry, which it is possible there may not be.